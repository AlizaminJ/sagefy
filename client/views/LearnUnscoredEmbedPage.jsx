import React from 'react'
import { number, string, shape } from 'prop-types'
import { convertUuidToUuid58 as to58 } from 'uuid58'
import Layout from './components/Layout'
import Icon from './components/Icon'

export default function LearnUnscoredEmbedPage({
  hash,
  progress,
  card: {
    entityId,
    name,
    data: { url },
  },
}) {
  return (
    <Layout
      hash={hash}
      page="LearnUnscoredEmbedPage"
      title="Learn"
      description="-"
      canonical={`/unscored-embed-cards/${to58(entityId)}`}
    >
      {progress && (
        <section>
          <progress value={progress} />
        </section>
      )}

      <section>
        <iframe src={url} width="600" height="400" title={name} />
      </section>

      <section>
        <form action="/next">
          <button type="submit">
            <Icon i="card" /> Next Card
          </button>
        </form>
      </section>
    </Layout>
  )
}

LearnUnscoredEmbedPage.propTypes = {
  hash: string.isRequired,
  progress: number,
  card: shape({
    name: string.isRequired,
    data: shape({
      url: string.isRequired,
    }).isRequired,
  }).isRequired,
}

LearnUnscoredEmbedPage.defaultProps = {
  progress: null,
}
