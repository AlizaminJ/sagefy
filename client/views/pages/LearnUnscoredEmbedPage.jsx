import React from 'react'
import { number, string, shape } from 'prop-types'
import Layout from '../components/Layout'
import Icon from '../components/Icon'

export default function LearnUnscoredEmbedPage({
  progress,
  card: {
    name,
    data: { url },
  },
}) {
  return (
    <Layout page="LearnUnscoredEmbedPage" title="Learn" description="-">
      {progress && (
        <section>
          <progress value={progress} />
        </section>
      )}

      <section>
        <iframe src={url} width="600" height="400" title={name} />
      </section>

      <section>
        <form action="/next">
          <button type="submit">
            <Icon i="card" /> Next Card
          </button>
        </form>
      </section>
    </Layout>
  )
}

LearnUnscoredEmbedPage.propTypes = {
  progress: number,
  card: shape({
    name: string.isRequired,
    data: shape({
      url: string.isRequired,
    }).isRequired,
  }).isRequired,
}

LearnUnscoredEmbedPage.defaultProps = {
  progress: null,
}
