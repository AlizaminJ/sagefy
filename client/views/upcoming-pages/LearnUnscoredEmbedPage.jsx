import React from 'react'
import Icon from '../components/Icon'

export default function LearnUnscoredEmbedPage() {
  return (
    <div className="LearnUnscoredEmbedPage">
      <section>
        <div className="meter">
          <div style={{ width: '95%' }} />
        </div>
      </section>

      <section>
        <iframe
          src="https://em.sagefy.org/examples/duration-and-rhythm"
          width="600"
          height="400"
          frameBorder="0"
          className="lift"
          title="Name of the embed card"
        />
      </section>

      <section>
        <form action="/choose-next">
          <button type="submit">
            <Icon i="card" /> Next Card
          </button>
        </form>
      </section>
    </div>
  )
}
