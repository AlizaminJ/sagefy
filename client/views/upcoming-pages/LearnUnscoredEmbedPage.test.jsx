import LearnUnscoredEmbedPage from './LearnUnscoredEmbedPage'

describe('LearnUnscoredEmbedPage', () => {
  it('should render LearnUnscoredEmbedPage', () => {
    expect(LearnUnscoredEmbedPage({})).toMatchSnapshot()
  })
})
