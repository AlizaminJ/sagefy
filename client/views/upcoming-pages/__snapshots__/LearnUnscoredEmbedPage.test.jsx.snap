// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnUnscoredEmbedPage should render LearnUnscoredEmbedPage 1`] = `
<div
  className="LearnUnscoredEmbedPage"
>
  <section>
    <progress
      value={0.7}
    />
  </section>
  <section>
    <iframe
      height="400"
      src="https://em.sagefy.org/examples/duration-and-rhythm"
      title="Name of the embed card"
      width="600"
    />
  </section>
  <section>
    <form
      action="/choose-next"
    >
      <button
        type="submit"
      >
        <Icon
          i="card"
          s="m"
        />
         Next Card
      </button>
    </form>
  </section>
</div>
`;
